src/api/queueMarkAsRead.js

```diff
@@ -30,12 +30,13 @@ const processQueue = (auth: Auth) => {
     return;
   }
 
-  messagesFlags(auth, unsentMessageIds, 'add', 'read');
-  unsentMessageIds = [];
+  messagesFlags(auth, unackedMessageIds, 'add', 'read').then(success => {
+    unackedMessageIds = unackedMessageIds.filter(id => !success.messages.includes(id));
```

Comment:
This is quadratic -- we go through the list `unackedMessageIds`, and for each of them we go through the list `success.messages` to look for the ID.

If there are thousands of IDs in the list, that could be significant. I think it's a bit tricky to make that happen -- because we only fetch 100 messages at a time it's tricky to have thousands of messages in the message list at once without having already scrolled through them, and if we're successfully fetching more as you scroll then we're probably successfully marking them off as read -- but it can happen.

And fortunately this only takes a line to fix: use `new Set` to make a JS `Set` from `success.messages`, and replace the `.includes` with checking that.